### ISSUE TITLE
secstorage.nfs.version is not used in two scenarios

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#9469]
the NFS version is not considered in two scenarios

- During systemvm template registration

https://github.com/apache/cloudstack/blob/7214c13e106368127c928b2028c80c3bacaf528c/engine/schema/src/main/java/com/cloud/upgrade/SystemVmTemplateRegistration.java#L87

https://github.com/apache/cloudstack/blob/7214c13e106368127c928b2028c80c3bacaf528c/engine/schema/src/main/java/com/cloud/upgrade/SystemVmTemplateRegistration.java#L436-L437

- When cleanup diagnosis files

https://github.com/apache/cloudstack/blob/7214c13e106368127c928b2028c80c3bacaf528c/server/src/main/java/org/apache/cloudstack/diagnostics/DiagnosticsServiceImpl.java#L479-L481

the nfs version is set to `null`

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~

~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by Revellski** [comment]
We can also not mount CD ISO to instances unless the NFS server explicitly supports NFSv3. This is with secstorage.nfs.version being set to `4.0` and `4.1`. 

```
Cloudstack 4.19.1.0
SSVM 4.19.1.0
Hypervisor XenServer
```

**Comment by DaanHoogland** [comment]
hm, I think this happened a few times now @JoaoJandre , me changing the setting of the milestone after you have, sorry. I think we can add fixes to 4.19 and merge forward, whereas enhancements should go in 4.20 directly, agree?

**Comment by JoaoJandre** [comment]
> hm, I think this happened a few times now @JoaoJandre , me changing the setting of the milestone after you have, sorry. I think we can add fixes to 4.19 and merge forward, whereas enhancements should go in 4.20 directly, agree?

@DaanHoogland Yeah, this makes sense.
Also, I took some time to realize we where editing the issues' labels/milestones at the same time xD.

**Comment by shwstppr** [comment]
@weizhouapache I've created #9559 for the original issues. Regarding https://github.com/apache/cloudstack/issues/9469#issuecomment-2260390777, do we any time mount ISOs on the hypervisor hosts?

**Comment by weizhouapache** [comment]
> > hm, I think this happened a few times now @JoaoJandre , me changing the setting of the milestone after you have, sorry. I think we can add fixes to 4.19 and merge forward, whereas enhancements should go in 4.20 directly, agree?
> 
> @DaanHoogland Yeah, this makes sense. Also, I took some time to realize we where editing the issues' labels/milestones at the same time xD.

I think there might be several checks , for example
- copy from primary to secondary (extract volume ? take volume snapshot ?)
- copy from secondary to primary (deploy vm from new template ? )
- mount configdrive ISO

**Comment by DaanHoogland** [comment]
> I think there might be several checks , for example
> 
>     * copy from primary to secondary (extract volume ? take volume snapshot ?)
> 
>     * copy from secondary to primary (deploy vm from new template ? )
> 
>     * mount configdrive ISO

Do we create new tickets for these @weizhouapache ? (and close this one!)

**Comment by weizhouapache** [comment]
> > I think there might be several checks , for example
> > ```
> > * copy from primary to secondary (extract volume ? take volume snapshot ?)
> > 
> > * copy from secondary to primary (deploy vm from new template ? )
> > 
> > * mount configdrive ISO
> > ```
> 
> Do we create new tickets for these @weizhouapache ? (and close this one!)

I did not check the scenarios, therefore cannot say if there are issues.
closing this ticket as the two issues reported in the description have been addressed by #9559 

thanks @shwstppr @DaanHoogland
